app/mailers/host_mailer.rb

```diff
@@ -14,39 +14,32 @@ def summary(options = {})
     total = 0
     total_metrics.values.each { |v| total += v }
 
-    subject = _("Puppet Summary Report - F:%{failed} R:%{restarted} S:%{skipped} A:%{applied} FR:%{failed_restarts} T:%{total}") % {
-      :failed => total_metrics["failed"],
-      :restarted => total_metrics["restarted"],
-      :skipped => total_metrics["skipped"],
-      :applied => total_metrics["applied"],
-      :failed_restarts => total_metrics["failed_restarts"],
-      :total => total
-    }
-
     @hosts = host_data
     @timerange = time
     @out_of_sync = hosts.out_of_sync
     @disabled = hosts.alerts_disabled
-
     set_url
-    set_locale_for user
 
-    mail(:to   => user.mail,
-         :from => Setting["email_reply_address"],
-         :subject => subject,
-         :date => Time.zone.now )
+    set_locale_for(user) do
+      subject = _("Puppet Summary Report - F:%{failed} R:%{restarted} S:%{skipped} A:%{applied} FR:%{failed_restarts} T:%{total}") % {
+        :failed => total_metrics["failed"],
+        :restarted => total_metrics["restarted"],
+        :skipped => total_metrics["skipped"],
+        :applied => total_metrics["applied"],
+        :failed_restarts => total_metrics["failed_restarts"],
+        :total => total
+      }
```

Comment:
Would it be faster to build the substitution hash outside of the block so it's done only once?